powerstore/resource_protection_policy.go

```diff
@@ -3,130 +3,187 @@ package powerstore
 	"errors"
+	"fmt"
 	"log"
 
 	"strings"
+	client "terraform-provider-powerstore/client"
 	"terraform-provider-powerstore/models"
 
 	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
 	"github.com/hashicorp/terraform-plugin-framework/attr"
-	"github.com/hashicorp/terraform-plugin-framework/diag"
-	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-type resourceProtectionPolicyType struct{}
+// newProtectionPolicyResource returns protection policy new resource instance
+func newProtectionPolicyResource() resource.Resource {
+	return &resourceProtectionPolicy{}
+}
+
+type resourceProtectionPolicy struct {
+	client *client.Client
+}
+
+// Metadata defines resource interface Metadata method
+func (r *resourceProtectionPolicy) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_protectionpolicy"
+}
 
-// GetSchema returns the schema for this resource.
-func (r resourceProtectionPolicyType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
-	return tfsdk.Schema{
-		Attributes: map[string]tfsdk.Attribute{
-			"id": {
-				Type:                types.StringType,
+// Schema defines resource interface Schema method
+func (r *resourceProtectionPolicy) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+
+		MarkdownDescription: "ProtectionPolicy resource",
+
+		Attributes: map[string]schema.Attribute{
+
+			"id": schema.StringAttribute{
 				Computed:            true,
 				Description:         "Unique identifier of the policy.",
 				MarkdownDescription: "Unique identifier of the policy.",
 			},
-			"name": {
-				Type:                types.StringType,
+
+			"name": schema.StringAttribute{
 				Required:            true,
 				Description:         "The name of the protection policy.",
 				MarkdownDescription: "The name of the protection policy.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
 			},
-			"description": {
-				Type:                types.StringType,
-				Computed:            true,
+
+			"description": schema.StringAttribute{
 				Optional:            true,
+				Computed:            true,
 				Description:         "The description of the protection policy.",
 				MarkdownDescription: "The description of the protection policy.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
 			},
-			"type": {
-				Type:                types.StringType,
-				Computed:            true,
+
+			"type": schema.StringAttribute{
```

Comment:
Is this attribute configurable by the user?

Reply:
No it is present only in GET API calls, removed optional from it